### PR TITLE
feat(tier4_external_api_msgs): add calibration status

### PR DESCRIPTION
**Pull request by TakumiKozaka-T4**
## Related Links

https://github.com/autowarefoundation/autoware.universe/pull/722

## Description

Add CalibrationStatus.msg and CalibrationDataRequest.srv. This is for cooperation with FMS.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Code is properly formatted
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code is properly formatted
- [x] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR**: Required to pass before the merge.
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: tier4_external_api_msgs/srv/CalibrationDataRequest.srv [hunk 1]
```diff
@@ -0,0 +1,5 @@
+bool data_request
```
Comment (isamu-takagi) [inline tier4_external_api_msgs/srv/CalibrationDataRequest.srv]: What is this data used for? If it is not needed, remove this line like the [Trigger.srv](https://github.com/ros2/common_interfaces/blob/master/std_srvs/srv/Trigger.srv).

File: tier4_external_api_msgs/srv/CalibrationDataRequest.srv [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+uint8[] graph_image
+float32[] accel_map
+float32[] brake_map
```
Comment (isamu-takagi) [inline tier4_external_api_msgs/srv/CalibrationDataRequest.srv]: The format of accel_map and brake_map is a CSV, so we need a 2D array or column size.
Moreover, the first column may be the string "default".

**Review by isamu-takagi — DISMISSED** [review]

File: tier4_external_api_msgs/msg/CalibrationStatus.msg [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#calibration target 
+uint8 ACCELBRAKEMAP=1
```
Comment (isamu-takagi) [inline tier4_external_api_msgs/msg/CalibrationStatus.msg]:
```suggestion
uint8 ACCEL_BRAKE_MAP=1
```

File: tier4_external_api_msgs/msg/CalibrationStatus.msg [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#calibration target
```
Comment (isamu-takagi) [inline tier4_external_api_msgs/msg/CalibrationStatus.msg]:
```suggestion
# calibration target 
```

File: tier4_external_api_msgs/msg/CalibrationStatus.msg [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#calibration target 
+uint8 ACCELBRAKEMAP=1
+
+#status constants
```
Comment (isamu-takagi) [inline tier4_external_api_msgs/msg/CalibrationStatus.msg]:
```suggestion
# status constants
```

File: tier4_external_api_msgs/msg/CalibrationStatus.msg [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+uint8 NORMAL=0
+uint8 WARNING=1
+uint8 ERROR=2
+uint8 STALE=3
+
+#fields
```
Comment (isamu-takagi) [inline tier4_external_api_msgs/msg/CalibrationStatus.msg]:
```suggestion
# fields
```

**Review by isamu-takagi — COMMENTED** [review]
Could you add a space for comments?

**Comment by isamu-takagi** [comment]
@TakumiKozaka-T4 
The name `CalibrationStatusStamped` seems a pair of time stamp and single `CalibrationStatus`,
so I think the following is preferred:
* `CalibrationStatus` + `CalibrationStatusArray`
* `CalibrationStatus` + `CalibrationStatuses`
* `CalibrationModuleStatus` + `CalibrationStatus`

**Review by yabuta — APPROVED** [review]
LGTM

**Pull request merged**